src/mscorlib/src/System/String.cs

```diff
@@ -2577,6 +2581,150 @@ public static int Compare(String strA, String strB, StringComparison comparisonT
             }                        
         }
 
+#if FEATURE_CORECLR
+        [Pure]
+        private unsafe Boolean StartsWithOrdinal(String startsWith)
+        {
+            var byteCount = startsWith.Length << 1;
+            // value.Length verified to be less than or equal to this.Length by calling function
+            if (byteCount > 512)
+            {
+                // incur transition to native costs if length > 512
+                return (nativeCompareOrdinalEx(this, 0, startsWith, 0, startsWith.Length) == 0);
+            }
+            
+            fixed (char* cpString = this)
+            fixed (char* cpStartsWith = startsWith)
+            {
+                var i = 0;
+                // using byte* cast to long* or char* to avoid 
+                // additions and multiplications of constant offsets in pointer arithmetic
+                var pString = (byte*)cpString;
+                var pStartsWith = (byte*)cpStartsWith;
+                var pExpected = pString;
+
+                if (byteCount >= 32)
+                {
+                    pExpected = pString + (byteCount & (~127));
+
+                    while (i < byteCount - 127)
+                    {
+                        i += 128;
+
+                        // Break is more efficent than return for unrolling
+                        // https://gist.github.com/adamsitnik/9d4f0107bdc15a802bbf#file-x86jit_break_vs_return_false
+                        if (*(long*)(pString) != *(long*)(pStartsWith)) break;
```

Comment:
`long` will perform poorly on 32-bit platforms. This needs to use int for !_WIN64. (You can use a trick with defining nint at the top of the file to avoid ifdefs everywhere.)